Tests/ApplicativeRouterTests/TestRouter.swift

```diff
@@ -0,0 +1,108 @@
+import ApplicativeRouter
+import Deriving
+import Either
+import Prelude
+
+enum Routes {
+  case root
+  case pathComponents(param: Either<String, Int>, commentId: Int)
+  case postBodyField(email: String)
+  case postBodyJsonDecodable(episode: Episode, param: Int)
+  case simpleQueryParams(ref: String?, active: Bool, t: Int)
+}
+
+let router: Router<Routes> = [
+
+  // /home
+  Routes.iso.root
+    <¢> get %> lit("home") %> end,
+
+  // /home/episodes/:string_or_int/comments/:int
+  Routes.iso.pathComponents
+    <¢> get %> lit("home") %> lit("episodes") %> pathParam(.intOrString) <%> lit("comments") %> .int <% end,
+
+  // POST /signup
+  Routes.iso.postBodyField
+    <¢> post %> formField("email") <% lit("signup") <% end,
+
+  // POST /episodes/:id
+  Routes.iso.postBodyJsonDecodable
+    <¢> post %> jsonBody(Episode.self) <%> lit("episodes") %> .int <% end,
+
+  // /path/to/somewhere/cool?ref=:optional_string&active=:bool&t=:int
+  Routes.iso.simpleQueryParams
```

Comment:
do you experience any compile time issues with long routes? for me swift compiler fails to compile more than five patterns in a row, i.e `"a" / "b" / "c" / "d" / "e" / "f"`. I wonder if it might be because I use a lot of variants of the same operator (`/`) with different generic types.